dotcom-rendering/src/model/buildLightboxImages.ts

```diff
@@ -0,0 +1,110 @@
+import type { EnhancedImageForLightbox, FEElement } from '../types/content';
+
+/**
+ * Only allow the lightbox to show images that have a source with a width greater
+ * than 620 pixels.
+ *
+ * Frontend has similar logic here:
+ * https://github.com/guardian/frontend/blob/126dfcbc1aa961650b7f7ff41ee50a12782bb62e/common/app/model/content.scala#L549
+ *
+ */
+const isLightboxable = (image: EnhancedImageForLightbox): boolean => {
+	const fields = image.media.allImages[0]?.fields;
+	if (!fields) return false; // Unlikely
+	const { width, height } = fields;
+	const orientation =
+		parseInt(width) > parseInt(height) ? 'horizontal' : 'portrait';
+	switch (orientation) {
+		case 'horizontal':
+			// If any width is above 620 we allow this image in lightbox
+			return image.media.allImages.some(
+				(mediaImg) => parseInt(mediaImg.fields.width) > 620,
+			);
+		case 'portrait':
+			// If any height is above 620 we allow this image in lightbox
+			return image.media.allImages.some(
+				(mediaImg) => parseInt(mediaImg.fields.height) > 620,
+			);
+	}
+};
+
+/**
+ * Generates a new array of lightbox images. Does not mutate.
+ *
+ * We decide this array, prior to rendering, because we create and
+ * insert the lightbox html in the dom at page load and to do this
+ * we need to know which images we want the lightbox to contain.
+ *
+ * This enhancer needs to run at a higher level to most other enhancers
+ * because it need both mainMediaElements and blocks in scope. Most other
+ * enhancers just have blocks in scope.
+ */
+export const buildLightboxImages = (
+	format: FEFormat,
+	blocks: Block[],
+	mainMediaElements: FEElement[],
+): EnhancedImageForLightbox[] => {
+	const images: EnhancedImageForLightbox[] = [];
+	mainMediaElements.forEach((element) => {
+		if (
+			element._type ===
+			'model.dotcomrendering.pageElements.ImageBlockElement'
+		) {
+			images.push(element);
+		}
+	});
+	blocks.forEach((block) => {
+		block.elements.forEach((element) => {
+			let imageElement: EnhancedImageForLightbox;
+			switch (element._type) {
+				case 'model.dotcomrendering.pageElements.ImageBlockElement': {
+					imageElement = element;
```

Comment:
Should this be `imageElement = { ...element }`, can't quite figure out if mutation is occurring here and/or if it's desired?

Reply:
I've added the spread just to be safe